src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -47,9 +50,27 @@ internal HttpHandlerToFilter(RTHttpBaseProtocolFilter filter)
             }
             cancel.ThrowIfCancellationRequested();
 
-
             RTHttpRequestMessage rtRequest = await ConvertRequestAsync(request).ConfigureAwait(false);
-            RTHttpResponseMessage rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+
+            RTHttpResponseMessage rtResponse;
+            try
+            {
+                rtResponse = await _next.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                throw;
+            }
+            catch (Exception)
+            {
+                Exception savedEx = (Exception)rtRequest.Properties[SavedExceptionLookupKey];
+                if (savedEx != null)
+                {
+                    throw savedEx;
```

Comment:
Looking at the MSDN, it seems I should save an ExceptionDispatchInfo object obtained by doing `ExceptionDispatchInfo.Capture(exception)` instead of saving an Exception like I'm doing now.  Then I can throw it later with `savedExceptionDispatchInfo.Throw()` ?